src/main/java/frc/robot/subsystems/Conveyor.java

```diff
@@ -139,6 +185,96 @@ public void stop() {
         back.stopMotor();
     }
 
+    @Override
+    public void periodic() {
+        previousColorSensorMeasurement = colorSensorMeasurement;
+        colorSensorMeasurement = colorSensor.getColorString();
+        lidarDistance = lidar.getAverageDistance();
+        colorSensorDistance = colorSensor.getProximity();
+
+        if (!doesBallExist()) {
+
+            frontState = ConveyorState.NONE;
+            backState = ConveyorState.NONE;
+
+        } else if (colorSensorMeasurement != previousColorSensorMeasurement) {
+
+            if (colorSensor.getColorString() == "Blue"
+                && MIN_COLOR_SENSOR_DISTANCE <= colorSensorDistance
+                && colorSensorDistance <= MAX_COLOR_SENSOR_DISTANCE) {
+
+                backState = frontState;
+                frontState = ConveyorState.BLUE;
+
+            } else if (colorSensor.getColorString() == "Red"
+                && MIN_COLOR_SENSOR_DISTANCE <= colorSensorDistance
+                && colorSensorDistance <= MAX_COLOR_SENSOR_DISTANCE) {
+
+                backState = frontState;
+                frontState = ConveyorState.RED;
+
+            } else if (!isBallQueued()) {
+
+                backState = frontState;
+                frontState = ConveyorState.NONE;
+
+            }
+
+        } else if (!isBallQueued()) {
```

Comment:
this is not a safe assumption to make. this code will cause the ball to be none on the second loop even if the ball is there beacuse the color sensor read the same color both times. if you checked the color sensor distance this *might* be ok